packages/bff/src/utilities/fileutils.ts

```diff
@@ -0,0 +1,209 @@
+  - occurrences: a map that contains the number of occurrences of a document name
+    (the same document name can be used multiple times in the same descriptor)
+  - uniqueNames: a map that contains the unique name for each document id
+*/
+export function buildFileDocumentRegistry(
+  eserviceDocuments: catalogApi.EServiceDoc[]
+): FileDocumentsRegistry {
+  return eserviceDocuments.reduce(
+    (fileRegistry: FileDocumentsRegistry, doc: catalogApi.EServiceDoc) => {
+      const originalName = doc.name;
+      const occurrence = fileRegistry.occurrences.get(doc.name) || 0;
+      fileRegistry.occurrences.set(originalName, occurrence + 1);
+
+      const parsedName = path.parse(originalName);
+      const newName = occurrence
+        ? `${parsedName.name}-${occurrence}${parsedName.ext}`
+        : originalName;
+
+      fileRegistry.uniqueNames.set(doc.id, newName);
+      return fileRegistry;
+    },
+    {
+      occurrences: new Map(),
+      uniqueNames: new Map(),
+    }
+  );
+}
+
+export function buildJsonConfig(
+  fileDocumentRegistry: FileDocumentsRegistry,
+  eservice: catalogApi.EService,
+  descriptor: catalogApi.EServiceDescriptor
+) {
+  return {
+    name: eservice.name,
+    description: eservice.description,
+    technology: eservice.technology,
+    mode: eservice.mode,
+    descriptor: {
+      interface: descriptor.interface && {
+        prettyName: descriptor.interface.prettyName,
+        path: descriptor.interface.path,
+      },
+      docs: descriptor.docs.map((doc) => {
+        const uniqueName = getUniqueNameByDocumentId(
+          fileDocumentRegistry,
+          doc.id
+        );
+        return {
+          prettyname: doc.prettyName,
+          path: `documents/${uniqueName}`,
+        };
+      }),
+      audience: descriptor.audience,
+      voucherLifespan: descriptor.voucherLifespan,
+      dailyCallsPerConsumer: descriptor.dailyCallsPerConsumer,
+      dailyCallsTotal: descriptor.dailyCallsTotal,
+      description: descriptor.description,
+      agreementApprovalPolicy: descriptor.agreementApprovalPolicy,
+      riskAnalysis: eservice.riskAnalysis.map((ra) => ({
+        name: ra.name,
+        riskAnalysisForm: {
+          version: ra.riskAnalysisForm.version,
+          singleAnswers: ra.riskAnalysisForm.singleAnswers.map((sa) => ({
+            key: sa.key,
+            value: sa.value,
+          })),
+          multiAnswers: ra.riskAnalysisForm.multiAnswers.map((ma) => ({
+            key: ma.key,
+            values: ma.values,
+          })),
+        },
+      })),
+    },
+  };
+}
+
+/* 
+  This function creates a zip file fetched from the S3 bucket 
+  using FileManager, the zip file containing the following files:
+  - descriptor's interface file
+  - descriptor's documents file
+  - configuration file
+
+  The zip folder structure in output is the following:
+  - zipFolderName
+      |
+      |- interface
+      |     |
+      |     |- interfaceFile.{fileExtension}
```

Comment:
According to this comment the interface file should be in a `/interface` subfolder, but I see it's added in the root like done for `configuration.json`. 
Looking at Scala it seems it's correct, consider fixing the comment?